src/lib/mrtg2/MRTG_lib.pm

```diff
@@ -1113,6 +1116,29 @@ sub cfgcheck ($$$$;$) {
             $error="yes";
         }
 
+        # Text color, format: #rrggbb
+        if (! defined $$rcfg{'textcolor'}{$rou}) {
+            # make some textcolor if not defined
+            # convert backg hex to decimal rgb
+            my @rgbvalue=split('',$$rcfg{'backgc'}{$rou});
+            my @rgb=(255,255,255);
+            @rgb=(hex($rgbvalue[1].$rgbvalue[2]),hex($rgbvalue[3].$rgbvalue[4]),hex($rgbvalue[5].$rgbvalue[6]));
+            # make contrast textcolor from backg for all backg values
+            if($rgb[0]+$rgb[1]+$rgb[2]<(256*3/2)) {
```

Comment:
Note: Brightness is calculated as the maximum (not the mean).

Reply:
This is subjective, you can use weighted mean. Blue-black combination is bad, blue-white combination is better.

Reply:
True, using https://en.wikipedia.org/wiki/HSL_and_HSV#Lightness as a reference, this is using the arithmetic mean, i.e. intensity. Whereas max is used in the HSV/HSB "hexcone" model. I guess the HSL "bi-hexcone" and luma models looks the nicest, but as you say it's subjective and something is better than nothing.

Reply:
while the 'magic' is nice, I think it would be a good thing to allow people to actually configure the color. Their milage might vary.